specification/playwrighttesting/PlaywrightTesting.Reporting/main.tsp

```diff
@@ -0,0 +1,783 @@
+
+  @doc("Number of retryCount attempts. Max retryCount is 100, after which testresults metadata will not be stored")
+  @visibility("read", "create")
+  @maxValue(100)
+  retryCount: int32;
+
+  @doc("Status of the test result. While uploading results, a single test execution status can't be Flaky")
+  @visibility("read", "create")
+  status: TestStatus;
+
+  @doc("Configuration for the web test.")
+  @visibility("read", "create")
+  webTestConfig?: WebTestConfig;
+
+  @doc("Continuous integration configuration.")
+  @visibility("read")
+  ciConfig?: CiConfig;
+
+  @doc("Duration of the test execution in milliseconds.")
+  durationInMs: int64;
+
+  @doc("Start time of the test execution.- (RFC 3339 literal format)")
+  startTime: utcDateTime;
+
+  @doc("Metadata related to attachments- comma separated IMAGE, VIDEO, TRACE as applicable")
+  attachmentsMetadata?: AttachmentKind[];
+
+  @doc("Previous retry summaries. While uploading results, each retry is uploaded separately.However, all the retries of a test gets grouped on the server side and can be retrieved as a single entity using GET API.")
+  @visibility("read")
+  previousRetries?: PreviousRetrySummary[];
+
+  @doc("List of tags associated with the test result.")
+  @visibility("read", "create")
+  @maxItems(100)
+  tags?: string[];
+
+  @doc("List of annotations (https://playwright.dev/docs/test-annotations).")
+  @visibility("read", "create")
+  @maxItems(100)
+  annotations?: string[];
+
+  @doc("Relative path in storage for all the artifacts of this test")
+  @visibility("read", "create")
+  artifactsPath?: string;
+}
+
+@doc("Supported browser Types for tests")
+union BrowserType {
+  string,
+
+  @doc("Chromium")
+  "Chromium",
+
+  @doc("Webkit")
+  "Webkit",
+
+  @doc("Firefox")
+  "Firefox",
+}
+
+@doc("Supported OS  for tests")
+union OsType {
+  string,
+
+  @doc("Windows")
+  "Windows",
+
+  @doc("Linux")
+  "Linux",
+
+  @doc("Macintosh")
+  "Mac",
+}
+
+@doc("Attachment kinds uploaded for a test result")
+union AttachmentKind {
+  string,
+
+  @doc("Screenshot")
+  "Image",
+
+  @doc("Video")
+  "Video",
+
+  @doc("Trace")
+  "Trace",
+}
+
+// Model representing web test configuration
+@doc("Configuration details for the web test.")
+model WebTestConfig {
+  @doc("Name of the CI job running the test.")
+  @maxLength(500)
+  jobName: string;
+
+  @doc("Name of the Playwright project.")
+  @maxLength(500)
+  projectName: string;
+
+  @doc("Name of the browser.")
+  browserType: BrowserType;
+
+  @doc("Operating system where the browser is hosted.")
+  os: OsType;
+}
+
+// Model representing summary of previous retryCount
+@doc("Summary of previous retries.")
+model PreviousRetrySummary {
+  @doc("Identifier of the test execution.")
+  testExecutionId: string;
+
+  @doc("Number of retryCount attempts.")
+  @maxValue(100)
+  retryCount: int32;
+
+  @doc("Status of the test execution.")
+  status: TestStatus;
+
+  @doc("Duration of the test execution.")
+  durationInMs: duration;
```

Comment:
I know we recommended changing this to the TypeSpec "duration" type, but sadly it does not currently support encoding to microseconds (Int64) on the wire, so for now it should say as Int. 
```suggestion
  @doc("Duration of the test execution in milliseconds.")
  durationInMs: int64;
```